### ISSUE TITLE
add args

### ISSUE THREAD

**Issue by mimooh** [Nyr/openvpn-install#798]
Lovely! :) Worked smoothly for me. How about further automation by adding args, e.g:

bash openvpn-install.sh --install --dns 8.8.8.8 
bash openvpn-install.sh --add "ClientName"
bash openvpn-install.sh --del "ClientName"

I get your reasoning, but I feel like installation should be separated from adding clients. I will probably add both ideas (separation and args) and might make a PR.

Another idea is adding google-authenticator support. I will need that for my project. I will try to script that and then share the code here. 

Great and useful project! :)
Karol

**Comment by Nyr** [comment]
First of all, thanks a lot for the donation!

While automation would be a nice to have feature, it will probably not be implemented in the short term for this project, honestly. And any change like that would also need to be prepared for [wireguard-install](https://github.com/Nyr/wireguard-install).

About the 2FA request, it is very unlikely to have this feature added to the main branch, ever. It is not a very popular request and it is not compatible with many client implementations.

My suggestion, if you need to adapt this project to your own needs, would be to create a fork from this repository. I have released this project with a very permissive license and there are no plans for structural changes at this stage, so merging future changes downstream is not going to be problematic at all.

Thanks also for your kind words about the project.

**Comment by mimooh** [comment]
I understand, I also wouldn't be happy with people messing up my code left
and right :)

Two comments then:

1. Before adding arguments, the code should be reorganized into functions.
Currently new-client is duplicated in your workflow and this approach would
solve it:
https://github.com/mimooh/openvpn-install/blob/master/openvpn-install.sh

2. Since you are not approving the idea :) I won't be pushing, I am ok with
that :)
Besides it's not really an issue that the server installer is interactive -
one is making that only once, no problem.
But, I am about to create hundreds of clients and these need to be
processed without dialog.
Therefore I will create a separate, short script for adding clients only in
my fork.
Then I'll see about 2FA, as a separate script as well.
Then I'll let you know and we will see if it fits your repo.

Regards,
Karol




Karol Kreński

**Comment by Nyr** [comment]
Sorry for the slow response, I am a bit busy nowadays.

Creating a short script to add the clients is a great idea for your use case, as this is simple functionality which can be independently implemented.

About the headless setup: I have also considered using variables instead, but probably arguments are more user-friendly. It is something which I'd like to do in the mid term, but not something which you should wait for at this time.

**Comment by mimooh** [comment]
I am glad with how my client went out, see ovclient.sh:
https://github.com/mimooh/ovclient

Still, google authenticator needs work, but since I need it for hundreds of
clients I will keep working on it.
Google authenticator is optional and if disabled the ovclient.sh works
correctly as far as I can tell.
Would you like the ovclient.sh being part of your project after I get
google authenticator fixed?
Or you prefer that it's a separate project as it is currently? Perhaps you
could then mention in your README there is such a tool?

Regards,
Karol Kreński

**Comment by Nyr** [comment]
It will not be merged into this project. Generally speaking I consider the project to be "feature-complete" and new features are very rarely added. Probably the only new feature which I am considering at this moment is custom DNS servers.

I understand that other people has specific needs and that is why the license allows users to do basically anything with the code without limitation, so anyone with particular needs can use the codebase as a starting point.

I understand that for some users the lack of niche features could be frustrating, but I really want to keep this simple.

**Comment by mimooh** [comment]
Which is good reasoning :) things should be simple. Okay then, I won't be
bothering you.

śr., 23 wrz 2020, 22:35 użytkownik Nyr <notifications@github.com> napisał:
